pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -426,11 +428,16 @@ func (h *NetworkUtilsHandler) CreateTapDevice(tapName string, queueNumber uint32
 	if err != nil {
 		return err
 	}
-	if err := tapDeviceSELinuxCmdExecutor.Execute(); err != nil {
+
+	// Device creation is retried due to https://bugzilla.redhat.com/1933627
+	// which has been observed on multiple occasions on CI runs.
+	const retryAttempts = 5
+	attempt, err := retry(tapDeviceSELinuxCmdExecutor.Execute, retryAttempts, time.Second)
```

Comment:
What is the reason for having the retry here and not inside the tapDeviceBuilder (surrounding the netlink call to create the device)? Here you create a new process on each retry. I find it  wasteful and I believe that the retry should be as close a possible to the "problemtic" code that should be retried.

Reply:
I do not know why we had to re-implement the tap device creation in our own specialized binary (virt-chroot) when this is already available though a platform binary (`ip`). I am uncomfortable to add there additional code just because I do not believe we should maintain such a thing.

The creation of an additional process is only expected in case this fails, and it is very (very) rare in the CI run and on production the chances for this to happen are even slimmer.
Considering that this is already fixed in new kernels, I would bet this will never happen in production.

Reply:
I don't find this reasoning as valid. Currently the tap device creation is in virt-chroot. We wrote and maintain this code. We shouldn't avoid touching it.

Reply:
> I don't find this reasoning as valid. Currently the tap device creation is in virt-chroot. We wrote and maintain this code. We shouldn't avoid touching it.

+1 

Also, I would bet that you'll find a lot less conflicts when eventually reverting this on the `virt-chroot` helper than in the huge network driver.

Reply:
> I don't find this reasoning as valid. Currently the tap device creation is in virt-chroot. We wrote and maintain this code. We shouldn't avoid touching it.

I guess that we disagree here.
I do not want to touch it because I want to drop it and use `ip`. Just because I do not want to maintain that code at all.

Telling me that given it is the current state I should accept it is not convincing me :)

> Also, I would bet that you'll find a lot less conflicts when eventually reverting this on the virt-chroot helper than in the huge network driver.

I have no plan to backport this.

Reply:
Unfortunately (to me), @qinqon and @AlonaKaplan gave a good reason why we cannot retry from the upper level at this stage. The virt-chroot command does actually two things in regards to the tap device. It creates it and then it sets the mtu of the link. Therefore, failure of the mtu setting will break this retry pattern.

I am forced to do this at the virt-chroot level, feeling really uncomfortable with it.
Hopefully we can find a way to do this better in the future.